connectors/src/resources/slack_configuration_resource.ts

```diff
@@ -111,14 +111,35 @@ export class SlackConfigurationResource extends BaseResource<SlackConfigurationM
     return new this(this.model, blob.get());
   }
 
-  async isBotWhitelistedToSummon(botName: string | string[]): Promise<boolean> {
-    return !!(await SlackBotWhitelistModel.findOne({
-      where: {
-        connectorId: this.connectorId,
-        botName: botName,
-        whitelistType: "summon_agent",
-      },
-    }));
+  async isBotWhitelistedToSummon(
+    botName: string | string[] | null,
```

Comment:
Here it would be good to:
1. migrate to just `string` not string | string[]
2. enforce a XOR botname/botid via the type (either one or the other, not both, nor neither)